### PR TITLE
Add instrumentation to content-app

### LINKED ISSUE #3829: Add the OpenTelemetry Aiohttp-server instrumentation as a vendor library
Until the [PR adding the aiohttp-server instrumentation support into the open-telemetry/opentelemetry-python-contrib library is  reviewed and accepted](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/1800)  I propose we vendor the code to enable `pulp-content` telemetry support.

We will remove it when `opentelemetry-python-contrib` support instrumenting the `aiohttp-server`.

### PR DESCRIPTION
**Pull request by lubosmj**
The instrumentation code from the upstream PR was brought down to pulpcore because of https://github.com/pypi/support/issues/3353.

closes #3829

### CODE REVIEW

**Comment by lubosmj** [comment]
Telemetry tests are enabled only for the Azure runner: https://github.com/pulp/pulpcore/blob/f0569f43fcbb689fbfbbc309fa7af4ee3518b346/template_config.yml#L52

File: pulpcore/tests/functional/api/pulp_file/test_telemetry_collection.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        url = urljoin(distribution.base_url, content_unit.relative_path)
+
+        s = requests.Session()
+        s.headers = {"User-Agent": test_path}
+
+        s.get(url, allow_redirects=False)
+        assert received_otel_span(
+            {
+                "http.method": "GET",
+                "http.target": f"/pulp/content/{distribution.base_path}/{file_name}",
```
Comment (lubosmj) [inline pulpcore/tests/functional/api/pulp_file/test_telemetry_collection.py:48]: I need to take care of `domains` and redirects to content storage here...

File: pulpcore/tests/functional/assets/otel_server.py [hunk 2]
```diff
@@ -68,6 +68,7 @@ async def _traces_handler(request):
                         for item in span.attributes
                     }
                     spans.append(attrs)
+                    print(attrs)
```
Comment (decko) [inline pulpcore/tests/functional/assets/otel_server.py:71]: Was intentional for this to be here?

**Review by decko — APPROVED** [review]
LGTM with just a question

File: pulpcore/tests/functional/assets/otel_server.py (hunk 2 above)
Comment (lubosmj) [reply pulpcore/tests/functional/assets/otel_server.py:71]: It is not cluttering the CI logs and it helps with further debugging to see whether the span was received on the server or not. Do you think logging is better than printing? I am going to replace it.

**Comment by bmbouter** [comment]
It seems the upstream is not yet able to release this, but they are working on it, see [this](https://github.com/open-telemetry/opentelemetry-python-contrib/issues/2053) and [this](https://pypi.org/search/?q=opentelemetry-instrumentation-aiohttp-server&o=).

Given that it's not available yet, I'm in favor of vendoring it for a release or two in Pulp because we have users who have been waiting and are very eager to try it. Also it comes with functional tests so unlike the previous pull request this does have tests that cover the vendored code.

@mdellweg wdyt about this approach? I don't want to have this merged without your input.

**Review by bmbouter — APPROVED** [review]
The tests look good, thanks for them! Also +1 to @decko 's question about the `print()` statement. Personally I'd remove it, but it's up to @lubosmj 

thanks @lubosmj !

File: pulpcore/tests/functional/assets/otel_server.py (hunk 2 above)
Comment (lubosmj) [reply pulpcore/tests/functional/assets/otel_server.py]: Removed.

**Pull request merged**